frontend/gpay_agent/lib/agent_views/welcome_agent.dart

```diff
@@ -1,87 +1,154 @@
-import 'package:agent_app/agent_datamodels/agent.dart';
-import 'package:agent_app/agent_datamodels/text_widget.dart';
+import 'dart:async';
+import 'dart:convert';
 import 'package:agent_app/agent_views/fetch_store.dart';
+import 'package:agent_app/custom_widgets//app_bar.dart';
+import 'package:agent_app/custom_widgets/personal_details_textbox.dart';
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter/painting.dart';
-import 'package:google_sign_in/google_sign_in.dart';
+import 'package:agent_app/agent_datamodels/globals.dart' as globals;
+import 'package:geolocator/geolocator.dart';
+import 'package:http/http.dart' as http;
+
+class WelcomeAgent extends StatefulWidget{
+  _WelcomeAgentState createState() => _WelcomeAgentState();
+}
 
 /// Builds the UI elements for Welcome Agent user interface and
-/// defines function for fetching number of store verified by agent.
-class WelcomeAgent extends StatelessWidget{
-  WelcomeAgent({this.googleSignIn, this.agent});
-  final GoogleSignIn googleSignIn;
-  final Agent agent;
+/// starts [positionSubscription] stream for continuously fetching agent location
+/// and make request to server for updating location in spanner.
+class _WelcomeAgentState extends State<WelcomeAgent>{
+  static const int UPDATE_DISTANCE = 1;
+  Geolocator _geolocator;
+  Position _position;
+  StreamSubscription<Position> positionSubscription;
+
+
+  /// Checks the status of GPS granted by user
+  /// locationAlways means GPS location will be traced when app is running either in background or foreground.
+  /// locationWhenInUse means GPS location will be traced when this application is the running app.
+  /// Default status will be granted if either of the two is granted
+  void checkPermission() {
+    _geolocator.checkGeolocationPermissionStatus().then((status) { print('status: $status'); });
+    _geolocator.checkGeolocationPermissionStatus(locationPermission: GeolocationPermission.locationAlways).then((status) { print('always status: $status'); });
+    _geolocator.checkGeolocationPermissionStatus(locationPermission: GeolocationPermission.locationWhenInUse)..then((status) { print('whenInUse status: $status'); });
```

Comment:
We probably need only one of these permissions.

Reply:
changed it